logic/src/commonMain/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepository.kt

```diff
@@ -205,26 +184,27 @@ internal class ConnectionDataSource(
     ) = wrapStorageRequest {
         connectionDAO.insertConnection(connectionMapper.modelToDao(connection))
     }.flatMap {
-        // This can fail, but the connection will be there and get synced in worst case scenario in next SlowSync
-        wrapApiRequest {
-            userDetailsApi.getUserInfo(idMapper.toApiModel(connection.qualifiedToId))
-        }.flatMap { userProfileDTO ->
-            wrapStorageRequest {
-                val selfUser = getSelfUser()
-                val userEntity = publicUserMapper.fromUserApiToEntityWithConnectionStateAndUserTypeEntity(
-                    userDetailResponse = userProfileDTO,
-                    connectionState = connectionStatusMapper.toDaoModel(state = connection.status),
-                    userTypeEntity = userTypeEntityTypeMapper.fromTeamAndDomain(
-                        otherUserDomain = userProfileDTO.id.domain,
-                        selfUserTeamId = selfUser.teamId?.value,
-                        otherUserTeamId = userProfileDTO.teamId,
-                        selfUserDomain = selfUser.id.domain,
-                        isService = userProfileDTO.service != null
+        selfTeamIdProvider().flatMap { teamId ->
+            // This can fail, but the connection will be there and get synced in worst case scenario in next SlowSync
+            wrapApiRequest {
+                userDetailsApi.getUserInfo(idMapper.toApiModel(connection.qualifiedToId))
+            }.flatMap { userProfileDTO ->
+                wrapStorageRequest {
+                    val userEntity = publicUserMapper.fromUserApiToEntityWithConnectionStateAndUserTypeEntity(
+                        userDetailResponse = userProfileDTO,
+                        connectionState = connectionStatusMapper.toDaoModel(state = connection.status),
+                        userTypeEntity = userTypeEntityTypeMapper.fromTeamAndDomain(
+                            otherUserDomain = userProfileDTO.id.domain,
+                            selfUserTeamId = teamId?.value,
+                            otherUserTeamId = userProfileDTO.teamId,
+                            selfUserDomain = selfUserId.domain,
+                            isService = userProfileDTO.service != null
+                        )
                     )
-                )
-
-                userDAO.insertUser(userEntity)
-                connectionDAO.updateConnectionLastUpdatedTime(connection.lastUpdate, connection.toId)
+                    userDAO.insertUser(userEntity)
+                    // todo: ask if we need to insert the conversation here when the connection is sent!
+                    connectionDAO.updateConnectionLastUpdatedTime(connection.lastUpdate, connection.toId)
```

Comment:
This I remember was to trigger the flow to update, so we can get it updated to the top of the conversations list.
But with the current view I don't know if is needed since we are getting all at once, user and connection =)